### DISCUSSION TITLE
Introducing Web Components in JavaScript Path

### DISCUSSION THREAD

**Discussion by elishamutang** [TheOdinProject/curriculum#28748] [Ideas]
Recently came across the concept of Web Components in vanilla JavaScript, and as the title says, maybe introducing Web Components somewhere near the end of the JavaScript course would be a good idea to provide students with some knowledge on creating reusable custom components. 

I reckon this would be a good segue when moving from JavaScript to React.

[Link to Web Components on MDN Web Docs](https://developer.mozilla.org/en-US/docs/Web/API/Web_components)

**Comment by mao-sz** [comment]
Thank you for the suggestion. Personally, I'm not inclined to have this included in the curriculum, even as a mention at the end of a course like the JS one before React. It's a *big* segue and rabbit hole, and one I don't think we need to suggest people go down during the curriculum. IMO it's much more than just some ideas and patterns, but entire new concepts and tools.

My feeling is that if this is to benefit anyone significantly, it'll be after TOP and for them to discover themselves - the curriculum needn't introduce it.

Paging @TheOdinProject/javascript for further thoughts.

**Reply by elishamutang** [reply]
Fair enough. I agree it's an entirely new concept which is more like a nice-to-have rather than a necessity since we have React. 

I guess prior to entering the React course I had the impression that React components would be very similar to custom web components. I guess in a way it is? But React makes it a lot easier to implement the use of components. 

I'm happy to scrap this idea. Thanks for the discussion! :)

**Reply by mao-sz** [reply]
I think this is a lot larger of a scope (more than just some patterns for structuring code) and also far out of scope of what TOP would benefit from (keeping a narrow focus), which is why we're more hesitant on "nice to haves" over "necessary"/"of significant benefit to the core purpose".

This is something where if we actually did want to include material on it, to be of good enough quality teaching, I'd think it'd need at the very least a whole section of lessons and projects using it, which is not something we actually would need in the current curriculum.

**Reply by elishamutang** [reply]
Yep, fair enough
